tls/extensions/s2n_server_key_share.c

```diff
@@ -92,18 +92,19 @@ static int s2n_server_key_share_recv(struct s2n_connection *conn, struct s2n_stu
 
     S2N_ERROR_IF(supported_curve_index < 0, S2N_ERR_ECDHE_UNSUPPORTED_CURVE);
 
+    struct s2n_ecc_evp_params* server_ecc_evp_params = &conn->secure.server_ecc_evp_params;
+    server_ecc_evp_params->negotiated_curve = ecc_pref->ecc_curves[supported_curve_index];
+
     /* If this is a HelloRetryRequest, we won't have a key share. We just have the selected group.
-     * Exit early so a proper keyshare can be generated. */
-    if (s2n_is_hello_retry_required(conn)) {
-        return 0;
+     * Set the server negotiated curve and exit early so a proper keyshare can be generated. 
```

Comment:
nit: This reads a little weird to me since nothing is set inside the if block. Could the "set" be removed? Or mention that we can exit early since the server's curve is already set?

Reply:
Will re-word.